index.d.ts

```diff
@@ -5,12 +5,15 @@ Symbol event names can be used to avoid name collisions when your classes are ex
 */
 type EventName = string | symbol;
 
-/**
-Emittery also accepts an array of strings and symbols as event names.
-*/
-type EventNames = EventName | readonly EventName[];
-
-declare class Emittery {
+// Helper type for turning the passed `EventData` type map into a list of string keys that don't require data alongside the event name when emitting. Uses the same trick that `Omit` does internally to filter keys by building a map of keys to keys we want to keep, and then accessing all the keys to return just the list of keys we want to keep.
+type DatalessEventNames<EventData> = {
+	[Key in keyof EventData]: EventData[Key] extends undefined ? Key : never;
+}[keyof EventData];
+
+declare class Emittery<
+	EventData = {[eventName: string]: any},
```

Comment:
```suggestion
	EventData = {[eventName: EventName]: any},
```

?

Reply:
Sadly TypeScript doesn't yet support symbols in index signatures, so this doesn't typecheck, otherwise I would. Since it's just the default, you can still override the EventData to support symbol event names. The TS team is gonna add support, it's just really hard apparently. https://github.com/microsoft/TypeScript/issues/1863

Reply:
At least add a TODO that it needs to be updated when that TS issue is fixed.